src/frontend/js/components/SearchFilterGroupModal/index.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useContext, useState } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import Modal from 'react-modal';
+
+import { fetchList } from '../../data/getResourceList/getResourceList';
+import { CourseSearchParamsContext } from '../../data/useCourseSearchParams/useCourseSearchParams';
+import { requestStatus } from '../../types/api';
+import { FilterDefinition, FilterValue } from '../../types/filters';
+import { Nullable } from '../../utils/types';
+import { useAsyncEffect } from '../../utils/useAsyncEffect';
+
+interface SearchFilterGroupModalProps {
+  filter: FilterDefinition;
+}
+
+const messages = defineMessages({
+  error: {
+    defaultMessage: 'There was an error while searching for {filterName}.',
+    description:
+      'Error message when the search for more filter value fails in the search filters modal.',
+    id: 'components.SearchFilterGroupModal.error',
+  },
+});
+
+export const SearchFilterGroupModal = ({
+  filter,
+}: SearchFilterGroupModalProps) => {
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+  const [values, setValues] = useState([] as FilterValue[]);
+  const [query, setQuery] = useState('');
+  const [error, setError] = useState(null as Nullable<
+    FormattedMessage.MessageDescriptor
+  >);
+
+  // We need the current course search params to get the facet counts
+  const [coursesSearchParams, dispatchCourseSearchParamsUpdate] = useContext(
+    CourseSearchParamsContext,
+  );
+
+  useAsyncEffect(async () => {
```

Comment:
When the modal is closed and the reopen, it has the same state. I don't know if this is want you want but you should maybe reset the state otherwise